CRM/Odoosync/Sync/ContactHandler.php

```diff
@@ -45,7 +45,30 @@ private function syncSingleContact() {
     $this->setLog(ts("Contact data:"));
     $this->setLog($sendData);
 
-    //TODO: in COS-17
+    $this->callOdooApi($sendData);
+  }
+
+  /**
+   * Sends contact's prepared data to Odoo API
+   *
+   * @param array $sendData
+   */
+  private function callOdooApi($sendData) {
+    $this->setLog(ts('Call Odoo Api ...'));
+
+    $login = CRM_Odoosync_Sync_Request_Auth::getInstance();
```

Comment:
I think this logic : 

```php
    $login = CRM_Odoosync_Sync_Request_Auth::getInstance();
    if ($login->odooUserId === FALSE) {
      $this->setLog(ts('Failed Odoo login.'));
      return;
    }
```

should be moved to CRM_Odoosync_Sync_Request_Contact constructor. 

and after doing that, I think you can move this logic : 

```php
    $syncContact = new CRM_Odoosync_Sync_Request_Contact($login->odooUserId);
    $syncResponse = $syncContact->sync($sendData);
    $this->setLog(ts('Odoo response:'));
    $this->setLog($syncResponse);
    $this->setLog(ts('End sync contact.'));
```

to syncSingleContact() method.